addon/mixins/e3-animated-child.js

```diff
@@ -62,6 +63,26 @@ export default Ember.Mixin.create(e3Child, {
   },
 
   /*
+   Apply the current state to the animation hash.
+   */
+  generateAnimationState() {
+    let animation = get(this, 'animation');
+    let attrs = this.get('attrs');
+    let resultState = {};
+
+    if(!animation) {
+      return resultState;
+    }
+
+    keys(animation).forEach((key, index) => {
+      let val = attrs.hasOwnProperty(key) ? this.getAttr(key) : get(animation, key);
+      resultState[key] = typeof val === 'function' ? val.call(this, data, index) : val;
```

Comment:
Ahhh — data should be `this.getAttr('data');` but it seems to have gotten lost in some refactoring. 

The thought is that if you create your own components, you could do something like this (In other words, the properties of your animation can depend on the data associated with that component):

```
export default CircleComponent.extend({
  animation: {
    duration(data) {
      return data.id % 2 === 0 ? 200 : 400;
    }
  }
});
```

So, items that have an odd `id` will have an animation duration of 400ms / even has a 200ms duration.